language/vm/vm_runtime/vm_runtime_types/src/native_functions/dispatch.rs

```diff
@@ -148,19 +151,49 @@ lazy_static! {
             vec![ByteArray, ByteArray],
             vec![ByteArray]
         );
+        // Event
+        add!(m, addr, "Event", "write_to_event_store",
+            |_| { NativeReturnStatus::InvalidArguments },
+            vec![ByteArray, U64, ByteArray],
+            vec![]
+        );
         // Vector
         add!(m, addr, "Vector", "length",
-            vector::native_length,
+            NativeVector::native_length,
             vec![Kind::All],
             vec![Reference(Box::new(tstruct(addr, "Vector", "T", vec![SignatureToken::TypeParameter(0)])))],
             vec![U64]
         );
-        // Event
-        add!(m, addr, "Event", "write_to_event_store",
-            |_| { NativeReturnStatus::InvalidArguments },
-            vec![ByteArray, U64, ByteArray],
+        add!(m, addr, "Vector", "new",
+            NativeVector::native_new,
+            vec![Kind::All],
+            vec![],
+            vec![tstruct(addr, "Vector", "T", vec![SignatureToken::TypeParameter(0)])]
+        );
+        add!(m, addr, "Vector", "get",
```

Comment:
Nit: this might be easier to review if there was a separate PR for the backing storage + `new` and the the new vector operations?